BUG: reset panic set masters and nodes

### Sealos Version

main

### How to reproduce the bug?

```
root@default-golang-0:~/go/src/github.com/labring/sealos# sealos reset --masters 192.168.64.86  --debug
are you sure to delete these nodes? Yes [y/yes], No [n/no] : y
2023-03-07T13:15:32 debug cluster info: apiVersion: apps.sealos.io/v1beta1
kind: Cluster
metadata:
  creationTimestamp: null
  name: default
spec:
  hosts:
  - ips:
    - 192.168.64.86:22
    roles:
    - master
    - arm64
  ssh:
    pk: /root/.ssh/id_rsa
    port: 22
status: {}

2023-03-07T13:15:32 debug sync cluster status is: apiVersion: apps.sealos.io/v1beta1
kind: Cluster
metadata:
  creationTimestamp: null
  deletionTimestamp: "2023-03-07T05:15:32Z"
  name: default
spec:
  hosts:
  - ips:
    - 192.168.64.86:22
    roles:
    - master
    - arm64
  ssh:
    pk: /root/.ssh/id_rsa
    port: 22
status: {}

2023-03-07T13:15:32 debug write reset cluster file to local: /root/.sealos/default/Clusterfile.1678166132
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/labring/sealos/pkg/types/v1beta1.(*Cluster).GetRootfsImage(0x40006cc480, {0x0, 0x0})
	github.com/labring/sealos/pkg/types/v1beta1/cluster_args.go:138 +0x194
github.com/labring/sealos/pkg/runtime.(*KubeadmRuntime).getKubeVersionFromImage(...)
	github.com/labring/sealos/pkg/runtime/runtime_getter.go:51
github.com/labring/sealos/pkg/runtime.(*KubeadmRuntime).Validate(0x40008b5900)
	github.com/labring/sealos/pkg/runtime/runtime.go:143 +0x44
github.com/labring/sealos/pkg/runtime.newKubeadmRuntime(0x40006cc480, 0x0, 0x0)
	github.com/labring/sealos/pkg/runtime/runtime.go:116 +0x15c
github.com/labring/sealos/pkg/runtime.NewDefaultRuntime(...)
	github.com/labring/sealos/pkg/runtime/runtime.go:128
github.com/labring/sealos/pkg/apply/processor.(*DeleteProcessor).Reset(0x400060f650?, 0x400060f801?)
	github.com/labring/sealos/pkg/apply/processor/delete.go:75 +0x3c
github.com/labring/sealos/pkg/apply/processor.DeleteProcessor.Execute({{0x38ea150?, 0x400060f650?}, {0x38e20e0?, 0x4000700240?}}, 0x38e20e0?)
	github.com/labring/sealos/pkg/apply/processor/delete.go:51 +0x324
github.com/labring/sealos/pkg/apply/applydrivers.(*Applier).deleteCluster(0x40000c2eb0)
	github.com/labring/sealos/pkg/apply/applydrivers/apply_drivers_default.go:242 +0x54
github.com/labring/sealos/pkg/apply/applydrivers.(*Applier).Delete(0x40000c2eb0)
	github.com/labring/sealos/pkg/apply/applydrivers/apply_drivers_default.go:233 +0xd0
github.com/labring/sealos/cmd/sealos/cmd.newResetCmd.func1(0x4000642300?, {0x40004d1ad0?, 0x3?, 0x3?})
	github.com/labring/sealos/cmd/sealos/cmd/reset.go:54 +0x9c
github.com/spf13/cobra.(*Command).execute(0x4000642300, {0x40004d1a70, 0x3, 0x3})
	github.com/spf13/cobra@v1.6.1/command.go:916 +0x5c4
github.com/spf13/cobra.(*Command).ExecuteC(0x4cc8700)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x340
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
github.com/labring/sealos/cmd/sealos/cmd.Execute()
	github.com/labring/sealos/cmd/sealos/cmd/root.go:47 +0x24
main.main()
	github.com/labring/sealos/cmd/sealos/main.go:27 +0x30
```

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Operating environment

```markdown
- Sealos version:
- Docker version:
- Kubernetes version:
- Operating system:
- Runtime environment:
- Cluster size:
- Additional information:
```


### Additional information

_No response_

--- a/pkg/types/v1beta1/cluster_args.go
+++ b/pkg/types/v1beta1/cluster_args.go
@@ -135,8 +135,10 @@ func (c *Cluster) GetRootfsImage(defaultMount string) *MountImage {
 		image = &MountImage{
 			Name:       fmt.Sprintf("%s-%d", c.Name, 0),
 			Type:       RootfsImage,
-			ImageName:  c.Spec.Image[0],
 			MountPoint: defaultMount,
+		}
+		if len(c.Spec.Image) > 0 {
+			image.ImageName = c.Spec.Image[0]
 		}
 	}
 	return image

